src/App.tsx

```diff
@@ -1,14 +1,82 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { FilterOptions } from './types/Filter';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [showModal, setIsModalVisible] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState<FilterOptions>(FilterOptions.All);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [iconSlash, setIconSlash] = useState<{ [key: number]: boolean }>({});
+
+  useEffect(() => {
+    setLoading(true);
+
+    getTodos()
+      .then(setTodos)
+      .finally(() => setLoading(false));
+  }, []);
+
+  const toggleModal = useCallback(
+    (todo?: Todo) => {
+      setIsModalVisible(!showModal);
+      setSelectedTodo(todo || null);
+
+      if (selectedTodo) {
+        setIconSlash(currentState => ({
+          ...currentState,
+          [selectedTodo.id]: false,
+        }));
+      }
+    },
+    [showModal, selectedTodo],
```

Comment:
The `useCallback` dependency array for `toggleModal` includes `selectedTodo`, which may not be necessary. Ensure that all dependencies are correctly listed to avoid unnecessary re-creations of the callback.